openfermioncirq/optimization/black_box.py

```diff
@@ -46,8 +46,21 @@ class BlackBox(metaclass=abc.ABCMeta):
             a list of tuples of the form (low, high), where low and high are
             lower and upper bounds on a parameter. The number of tuples
             should be equal to the dimension of the black box.
+        cost_of_evaluate: If specified, then calls to `evaluate` will be
+            redirected to `evaluate_with_cost` with the specified cost.
     """
 
+    def __init__(self,
+                 cost_of_evaluate: Optional[float]=None,
+                 **kwargs) -> None:
+        """
+        Args:
+            cost_of_evaluate: An optional cost associated with the
+                `evaluate` method. If specified, the `evaluate` method
+                will defer to `evaluate_with_cost` with the specified cost.
+        """
+        self.cost_of_evaluate = cost_of_evaluate
+
     @abc.abstractproperty
```

Comment:
My preference is that if you are going to use advanced features of the language like this you make a special effort to add comments explaining what you're doing. For instance, why are you making this an abstract property?

Reply:
I think you're underestimating our users' willingness to learn to use common Python features.

Reply:
How many of our users do you know? Because I know many and I assure you that you are overestimating their willingness.